backend/icpy/api/rest_api.py

```
@@ -484,6 +491,70 @@ async def get_file_content(path: str):
+                abs_path = None
+                for cand in candidates:
+                    if os.path.exists(cand):
+                        abs_path = cand
+                        break
+
+                if abs_path is None or os.path.isdir(abs_path):
+                    raise HTTPException(status_code=404, detail="File not found")
+
+                # Security: ensure selected path is inside either fs_root or workspace_root
+                if not (abs_path.startswith(fs_root) or abs_path.startswith(workspace_root)):
+                    raise HTTPException(status_code=400, detail="Path outside workspace root")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)